internal/provider/virtualmachine/resource_virtualmachine_constructor.go

```diff
@@ -47,6 +49,38 @@ func (c *Constructor) Setup() util.Processors {
 				return nil
 			},
 		},
+		{
+			Field: constants.FieldVirtualMachineEFI,
+			Parser: func(i interface{}) error {
+				domain := vmBuilder.VirtualMachine.Spec.Template.Spec.Domain
+				if i.(bool) {
+					domain.Firmware = &kubevirtv1.Firmware{
+						Bootloader: &kubevirtv1.Bootloader{
+							EFI: &kubevirtv1.EFI{
+								SecureBoot: pointer.Bool(false),
+							},
+						},
+					}
+				} else {
+					domain.Firmware = nil
+				}
+				vmBuilder.VirtualMachine.Spec.Template.Spec.Domain = domain
+				return nil
+			},
+		},
+		{
+			Field: constants.FieldVirtualMachineSecureBoot,
+			Parser: func(i interface{}) error {
+				if i.(bool) {
+					firmware := vmBuilder.VirtualMachine.Spec.Template.Spec.Domain.Firmware
+					if firmware == nil || firmware.Bootloader == nil || firmware.Bootloader.EFI == nil {
+						return errors.New("EFI must be enabled to use Secure Boot。")
+					}
+					firmware.Bootloader.EFI.SecureBoot = pointer.Bool(true)
```

Comment:
does change the var `firmware` will effect  `vmBuilder.VirtualMachine.Spec.Template.Spec.Domain.Firmware` ?

Reply:
```go
{
			Field: constants.FieldVirtualMachineEFI,
			Parser: func(i interface{}) error {
				var firmware *kubevirtv1.Firmware
				if i.(bool) {
					firmware = &kubevirtv1.Firmware{
						Bootloader: &kubevirtv1.Bootloader{
							EFI: &kubevirtv1.EFI{
								SecureBoot: pointer.Bool(false),
							},
						},
					}
				}
				vmBuilder.VirtualMachine.Spec.Template.Spec.Domain.Firmware = firmware
				return nil
			},
			Required: true,
		},
		{
			Field: constants.FieldVirtualMachineSecureBoot,
			Parser: func(i interface{}) error {
				firmware := vmBuilder.VirtualMachine.Spec.Template.Spec.Domain.Firmware
				if firmware == nil || firmware.Bootloader == nil || firmware.Bootloader.EFI == nil {
					return errors.New("EFI must be enabled to use Secure Boot. ")
				}
				firmware.Bootloader.EFI.SecureBoot = pointer.Bool(true)
				vmBuilder.VirtualMachine.Spec.Template.Spec.Domain.Firmware = firmware
				return nil
			},
		},
```